--- a/cerbero/recipes/gst-omx-1.0.recipe	2020-12-05 12:16:50.680144258 +0000
+++ b/cerbero/recipes/gst-omx-1.0.recipe	2020-12-05 14:39:43.476000000 +0000
@@ -21,5 +21,15 @@
             self.append_env('CPPFLAGS', *flags)
             self.append_env('LDFLAGS', '-L/opt/vc/lib')
             self.meson_options['target'] = 'rpi'
+            self.meson_options['header_path'] = '/opt/vc/include/IL'
         else:
             self.meson_options['target'] = 'generic'
+
+    async def configure(self):
+        if self.config.variants.rpi:
+            try:
+                os.system('patch -p1 < recipes/libgstomx/0001_gstomx.pc.in.patch')
+                os.system('patch -p1 < recipes/libgstomx/0002_meson.build.patch')
+            except:
+                pass
+        await super(Recipe, self).configure()
