--- a/cerbero/cerbero/enums.py	2021-09-26 19:22:41.151754459 +0100
+++ b/cerbero/cerbero/enums.py	2021-09-26 19:04:28.100627746 +0100
@@ -37,6 +37,7 @@
     X86_64 = 'x86_64'
     UNIVERSAL = 'universal'
     ARM = 'arm'
+    ARMv6L = 'armv6l'
     ARMv7 = 'armv7'
     ARMv7S = 'armv7s'
     ARM64 = 'arm64'
@@ -47,12 +48,12 @@
         Note that it will include 32bit *and* 64bit ARM targets. If you
         wish to do something special for 64bit you should first check for
         that before calling this method.'''
-        return arch in [Architecture.ARM, Architecture.ARMv7,
+        return arch in [Architecture.ARM, Architecture.ARMv6L, Architecture.ARMv7,
                 Architecture.ARMv7S, Architecture.ARM64]
 
     @staticmethod
     def is_arm32(arch):
-        return arch in [Architecture.ARM, Architecture.ARMv7, Architecture.ARMv7S]
+        return arch in [Architecture.ARM, Architecture.ARMv6L, Architecture.ARMv7, Architecture.ARMv7S]
 
 
 class Distro:
