--- a/cerbero/recipes/gst-plugins-good-1.0.recipe	2020-12-24 07:46:14.551838213 +0000
+++ b/cerbero/recipes/gst-plugins-good-1.0.recipe	2021-02-07 13:42:52.539158541 +0000
@@ -271,3 +271,16 @@
 
         if self.config.variants.nodebug:
             self.append_env('CFLAGS', '-DGST_LEVEL_MAX=GST_LEVEL_FIXME')
+
+    async def configure(self):
+        if self.config.variants.rpi:
+            try:
+                os.system('patch -p1 < recipes/gst-plugins-good-1.0/0001_RaspiCapture.c.patch')
+                os.system('patch -p1 < recipes/gst-plugins-good-1.0/0002_gstrpicamsrc.c.patch')
+                os.system('patch -p1 < recipes/gst-plugins-good-1.0/0003_gstmultifilesink.c.patch')
+                os.system('patch -p1 < recipes/gst-plugins-good-1.0/0004_gstmultifilesink.h.patch')
+                os.system('patch -p1 < recipes/gst-plugins-good-1.0/0005_RaspiCapture.h.patch')
+            except:
+                pass
+        await super(Recipe, self).configure()
+
